src/dreyfus_index.erl

```diff
@@ -157,9 +157,26 @@ handle_call(info, _From, State) -> % obsolete
     Reply = info_int(State#state.index_pid),
     {reply, Reply, State}.
 
+handle_cast({ddoc_updated, DDocResult},
+            #state{
+                index=#index{sig=Sig, dbname=DbName}=Index
+            }=State) ->
+    case check_if_index_is_deleted(DbName, DDocResult, Sig) of
+        true ->
+            erlang:send_after(index_shutdown_delay(), self(), index_deleted);
```

Comment:
this will crash the gen server as it returns a value that handle_cast does not expect (a TimerRef).